web/src/activity.ts

```diff
@@ -8,6 +8,29 @@ import {page_params} from "./page_params";
 import * as presence from "./presence";
 import * as watchdog from "./watchdog";
 
+// introduced a new data structure to store the presence data of users for the last 10 years.
+export const user_setting_presence_info: Record<
+    string,
+    {active_timestamp: number; idle_timestamp: number}
+> = {};
+
+// Define the schema for the response of the POST request to fetch the presence data of last 10 years.
+const post_presence_user_settings_response_scheme = z.object({
+    result: z.string(),
+    msg: z.string(),
+    presences: z
+        .record(
+            z.string(),
+            z.object({
+                active_timestamp: z.number(),
+                idle_timestamp: z.number(),
+            }),
+        )
+        .optional(),
+    server_timestamp: z.number().optional(),
+    presence_last_update_id: z.number().optional(),
+});
```

Comment:
I don't think you have to duplicate this schema, you should probably just be able to re-use `post_presence_response_schema` here?

Reply:
Thanks for pointing out this..I will remove this redundant part.